core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -106,13 +111,68 @@ public Void call() throws S3Exception {
     });
   }
 
+  /**
+   * @summary deletes a bucket
+   * @param bucket the Alluxio path
+   * @return the response object
+   */
+  @DELETE
+  @Path(BUCKET_PARAM)
+  @ReturnType("java.lang.Void")
+  public Response deleteBucket(@PathParam("bucket") final String bucket) {
+    return S3RestUtils.call(bucket, new S3RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws S3Exception {
+        String bucketPath =  AlluxioURI.SEPARATOR + bucket;
+        if (bucketPath.contains(BUCKET_SEPARATOR)) {
```

Comment:
this block is identical to that in createBucket, can be put into a helper method